javaslang/src/main/java/javaslang/control/Option.java

```diff
@@ -54,14 +59,14 @@
      */
     static <T> Option<Seq<T>> sequence(Iterable<? extends Option<? extends T>> values) {
         Objects.requireNonNull(values, "values is null");
-        List<T> list = List.empty();
+        Stream<T> stream = Stream.empty();
         for (Option<? extends T> value : values) {
             if (value.isEmpty()) {
                 return Option.none();
             }
-            list = list.prepend(value.get());
+            stream = stream.append(value.get());
         }
-        return Option.some(list.reverse());
+        return Option.some(stream);
```

Comment:
why is this an `Option` exactly, when it could simply return an empty stream instead?
@danieldietrich?